provider/openstack/local_test.go

```diff
@@ -881,7 +884,10 @@ func (s *localServerSuite) TestStartInstanceDeletesSecurityGroupsOnFailure(c *gc
 	_, _, _, err := testing.StartInstance(env, s.callCtx, s.ControllerUUID, "100")
 	c.Assert(err, gc.NotNil)
 
-	assertSecurityGroups(c, env, []string{"default"})
+	assertSecurityGroups(c, env, []string{
+		"juju-deadbeef-1bad-500d-9000-4b1d0d06f00d-deadbeef-0bad-400d-8000-4b1d0d06f00d",
```

Comment:
**suggestion:** maybe using the value from initialization would make this test more maintainable:

```go

	assertSecurityGroups(c, env, []string{
		"juju-" + s.Tests.ControllerUUID,
		"default",
	})
```
(it's explicit that we want to keep default security group and the one suffixed by the controller UUID)

Reply:
Yes fair. There's quite a few cases of this string being hardcoded in this file, I'll refactor them out

Reply:
Done

Reply:
**thought:** That's good to me, even I am not sure that introducing a global-scope variable for that is the better way. Keeping most of the information in the test itself is better. The question is, what is a relevant information ? I don't have enough context to answer here.